mono/metadata/domain-internals.h

```diff
@@ -235,11 +235,16 @@ struct _MonoJitInfo {
 
 #define MONO_SIZEOF_JIT_INFO (offsetof (struct _MonoJitInfo, clauses))
 
+typedef struct {
+	gpointer *static_data;
+	uint32_t gc_handle;
+} ContextStaticData;
+
 struct _MonoAppContext {
 	MonoObject obj;
 	gint32 domain_id;
 	gint32 context_id;
-	gpointer *static_data;
+	ContextStaticData *data;
```

Comment:
OK, but how would this work exactly? When we update the `gpointer static_data` field in `MonoAppContext`, we would also need to update the equivalent field in `ContextStaticData` so the reference queue callback actually has something to clean up. But we can't do that without having a pointer to the `ContextStaticData`. So storing it there makes sense to me.